k4MarlinWrapper/examples/event_display.py

```diff
@@ -18,58 +18,53 @@
 # limitations under the License.
 #
 
-from Gaudi.Configuration import *
+import sys
+
+from Gaudi.Configuration import INFO
+from Configurables import MarlinProcessorWrapper, k4DataSvc, PodioInput, EDM4hep2LcioTool, GeoSvc
+from k4FWCore.parseArgs import parser
+from k4MarlinWrapper.inputReader import create_reader, attach_edm4hep2lcio_conversion
+
+
+parser.add_argument(
+    "--inputFiles",
+    action="extend",
+    nargs="+",
+    metavar=["file1", "file2"],
+    help="One or multiple input files",
+)
+parser.add_argument(
+    "--compactFile", help="Compact detector file to use", type=str, default=""
+)
+
+reco_args = parser.parse_known_args()[0]
 
-from Configurables import MarlinProcessorWrapper, k4DataSvc, PodioInput, EDM4hep2LcioTool
 algList = []
+svcList = []
 
-evtsvc = k4DataSvc('EventDataSvc')
-evtsvc.input = ''
+evtsvc = k4DataSvc("EventDataSvc")
+svcList.append(evtsvc)
 
-inp = PodioInput('InputReader')
-inp.collections = [
-  'MCParticles',
-  'VertexBarrelCollection',
-  'VertexEndcapCollection',
-  'InnerTrackerBarrelCollection',
-  'OuterTrackerBarrelCollection',
-  'InnerTrackerEndcapCollection',
-  'OuterTrackerEndcapCollection',
-  'ECalEndcapCollection',
-  'ECalEndcapCollectionContributions',
-  'ECalBarrelCollection',
-  'ECalBarrelCollectionContributions',
-  'ECalPlugCollection',
-  'ECalPlugCollectionContributions',
-  'HCalBarrelCollection',
-  'HCalBarrelCollectionContributions',
-  'HCalEndcapCollection',
-  'HCalEndcapCollectionContributions',
-  'HCalRingCollection',
-  'HCalRingCollectionContributions',
-  'YokeBarrelCollection',
-  'YokeBarrelCollectionContributions',
-  'YokeEndcapCollection',
-  'YokeEndcapCollectionContributions',
-  'LumiCalCollection',
-  'LumiCalCollectionContributions',
-  'BeamCalCollection',
-  'BeamCalCollectionContributions',
-]
+if reco_args.compactFile:
+    compact_file = reco_args.compactFile
+else:
+    # add the same default as before
+    # TODO: maybe modernize this
+    compact_file = "CLICPerformance/Visualisation/CLIC_o3_v06_CED/CLIC_o3_v06_CED.xml"
 
-MyInitializeDD4hep = MarlinProcessorWrapper("MyInitializeDD4hep")
-MyInitializeDD4hep.OutputLevel = INFO
-MyInitializeDD4hep.ProcessorType = "InitializeDD4hep"
-MyInitializeDD4hep.Parameters = {
-                                 "DD4hepXMLFile": ["CLICPerformance/Visualisation/CLIC_o3_v06_CED/CLIC_o3_v06_CED.xml"]
-                                 }
+geoSvc = GeoSvc("GeoSvc")
+geoSvc.detectors = [compact_file]
+geoSvc.OutputLevel = INFO
+geoSvc.EnableGeant4Geo = False
+svcList.append(geoSvc)
 
-MyEventSelector = MarlinProcessorWrapper("MyEventSelector")
-MyEventSelector.OutputLevel = INFO
-MyEventSelector.ProcessorType = "EventSelector"
-MyEventSelector.Parameters = {
-                              "EventList": ["28", "0", "33", "0", "52", "0", "63", "0", "73", "0", "78", "0"]
-                              }
+
+if reco_args.inputFiles:
```

Comment:
This should be impossible to be false with the `nargs="+"`, I think. On a slightly more "philosophical level", do we even want an event display without any events?

Reply:
It is still possible :( but I guess I can make it a required argument.
(which gets us into a different philosophical discussion, but I am fine with doing it for this and for the compact file)

Reply:
urgh but it breaks `--help` then

Reply:
Ah right, because we go through the whole `k4run` machinerie as well. Then having `--help` working is the better option.